src/sentry/search/events/datasets/metrics.py

```diff
@@ -60,6 +61,13 @@ def resolve_metric(self, value: str) -> int:
         self.builder.metric_ids.add(metric_id)
         return metric_id
 
+    def resolve_tag_value(self, value: str) -> Union[str, int]:
+        if self.builder.is_performance and options.get(
+            "sentry-metrics.performance.tags-values-are-strings"
+        ):
+            return value
+        return self.resolve_value(value)
+
     def resolve_value(self, value: str) -> int:
```

Comment:
cause `resolve_value` was effectively `resolve_tag_value` since the only difference was that we wanted to store metric_ids for the WHERE clause when using `resolve_metric`